packages/lesswrong/components/hooks/useAuth0Client.ts

```diff
@@ -0,0 +1,75 @@
+import { auth0ClientSettings } from "../../lib/publicSettings";
+import { combineUrls } from "../../lib/vulcan-lib";
+
+class Auth0ClientError extends Error {
+  constructor(message?: string | null) {
+    super(message || "Something went wrong");
+  }
+}
+
+class Auth0Client {
+  private async post(endpoint: string, body: JsonRecord): Promise<void> {
+    const result = await fetch(endpoint, {
+      method: "POST",
+      body: JSON.stringify(body),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (result.status !== 200) {
+      const data = await result.json();
+      let message = data.error || data.error_description || "Something went wrong";
+      // Remove the ugly full-stop that auth0 appends to its error messages
+      if (message[message.length - 1] === ".") {
+        message = message.slice(0, -1);
+      }
+      throw new Auth0ClientError(message);
+    }
+  }
+
+  async login(email: string, password: string): Promise<void> {
+    await this.post("/auth/auth0/embedded-login", {email, password});
+  }
+
+  async signup(email: string, password: string): Promise<void> {
+    await this.post("/auth/auth0/embedded-signup", {email, password});
+  }
+
+  socialLogin(connection: "google-oauth2" | "facebook"): void {
```

Comment:
Social logins are still handled using the same Auth0 "universal login" page, but we pass in the `connection` parameter which tells it to redirect us straight to google or facebook to continue logging in.